src/backend/distributed/catalog/dependency.c

```diff
@@ -101,6 +112,52 @@ SupportedDependencyByCitus(const ObjectAddress *address)
 			return true;
 		}
 
+		case OCLASS_TYPE:
+		{
+			switch (get_typtype(address->objectId))
+			{
+				case TYPTYPE_ENUM:
+				case TYPTYPE_COMPOSITE:
+				{
+					return true;
+				}
+
+				case TYPTYPE_BASE:
+				{
+					/*
+					 * array types should be followed but not created, as they get created
+					 * by the original type.
+					 */
+					return type_is_array(address->objectId);
+				}
+
+				default:
+				{
+					/* type not supported */
```

Comment:
I wonder whether we should warn the user that we're not going to support the other types. For example, see the following:

```SQL
postgres=# CREATE TYPE floatrange AS RANGE (
postgres(#     subtype = float8,
postgres(#     subtype_diff = float8mi
postgres(# );
CREATE TYPE
Time: 2.742 ms
postgres=# -- we don't support
postgres=# CREATE TABLE test_table_2(a int, c floatrange);
CREATE TABLE
Time: 5.713 ms
postgres=# SELECT create_distributed_table('test_table_2', 'a');
ERROR:  type "public.floatrange" does not exist
CONTEXT:  while executing command on localhost:9702
Time: 30.242 ms
```

As a user who has distributed many types which worked well, seeing this error was confusing. I can argue that WARNING would also be confusing. For example, I remember that creating schemas  was very confusing to the user before https://github.com/citusdata/citus/pull/665. So, I cannot push for a warning either.

So, maybe one step back, is there a reason that we do not support range types?

Reply:
good question, range types rely on functions and operators which we don't support at this moment but is about to be solved as well. Then we definitely should push for range types (and operators) to be added.

I have been thinking about adding a warning when we do the dependency traversal and run into an unsupported dependency. But I think that would give too many warnings at the moment.

That said, the primary reason is that ENUM and COMPOSITE types were the primary scope for this PR, so I haven't looked deep at range types. Can you actually use PLPGSQL functions, or are these only C functions that are supported?

Looking at the docs I read this about one of the functions
```
Creating a canonical function is a bit tricky, ... To do this, you must first create a shell type
```

Which comes close to a base type. This causes circular dependencies which are not covered at the moment and will probably stay a hard problem as we need to generate the right commands in the right order:
 - shell type
 - distribute function
 - upgrade type
 Not impossible, but definitely not easy.

Reply:
> Can you actually use PLPGSQL functions, or are these only C functions that are supported?

It looks like we can use plpgsql functions:

```SQL
 CREATE OR REPLACE FUNCTION onder_float8mi(float, float)
 RETURNS float AS 
 $$
 BEGIN
    RETURN $1 * $2;
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;

CREATE TYPE onder_floatrange AS RANGE (
   subtype = float8,
   subtype_diff = onder_float8mi
 );

```

Reply:
lets punt range times to when we have functions supported.
most of the code is already in place now so could be a nice task for a new joiner